ui/components/multichain/asset-picker-amount/asset-picker-amount.tsx

```diff
@@ -1,290 +1,100 @@
-export const AssetPickerAmount = () => {
+export const AssetPickerAmount = ({
+  asset,
+  amount,
+  onAmountChange,
+  ...assetPickerProps
+}: AssetPickerAmountProps) => {
   const t = useI18nContext();
-  const { asset, amount } = useSelector(getCurrentDraftTransaction);
+
   const selectedAccount = useSelector(getSelectedInternalAccount);
 
-  const { error } = amount;
+  const isFiatPrimary = useSelector(getIsFiatPrimary);
 
-  const balanceColor = error
-    ? TextColor.errorDefault
-    : TextColor.textAlternative;
+  const [isFocused, setIsFocused] = useState(false);
 
-  const renderAssetDisplay = () => {
-    if (asset.type === AssetType.native) {
-      return (
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        // @ts-ignore: Other props are optional but the compiler expects them
-        <UserPreferencedCurrencyDisplay
-          value={asset.balance}
-          type={PRIMARY}
-          textProps={{
-            color: balanceColor,
-            variant: TextVariant.bodySm,
-          }}
-          suffixProps={{
-            color: balanceColor,
-            variant: TextVariant.bodySm,
-          }}
-        />
-      );
-    }
-    if (
-      asset.details?.standard === TokenStandard.ERC20 ||
-      asset.details?.standard === TokenStandard.ERC721
-    ) {
-      return (
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        // @ts-ignore: Details should be defined for token assets
-        <TokenBalance
-          token={asset.details}
-          textProps={{
-            color: balanceColor,
-            variant: TextVariant.bodySm,
-          }}
-          suffixProps={{
-            color: balanceColor,
-            variant: TextVariant.bodySm,
-          }}
-        />
-      );
-    } else if (asset.details?.standard === TokenStandard.ERC1155) {
-      const {
-        details: { balance },
-      } = asset as any;
-      return (
-        <Text color={balanceColor} marginRight={1} variant={TextVariant.bodySm}>
-          {balance}
-        </Text>
-      );
-    }
-    return null;
-  };
+  const { error } = amount;
 
   useEffect(() => {
     if (!asset) {
       throw new Error('No asset is drafted for sending');
     }
   }, [selectedAccount]);
 
+  let borderColor = BorderColor.borderDefault;
+
+  if (amount.error) {
+    borderColor = BorderColor.errorDefault;
+  } else if (isFocused) {
+    borderColor = BorderColor.primaryDefault;
+  }
+
   return (
     <Box className="asset-picker-amount">
       <Box display={Display.Flex}>
-        <Label>
-          {asset.type === AssetType.NFT ? t('asset') : t('amount')}:
-        </Label>
-        <MaxClearButton asset={asset} />
+        <Label variant={TextVariant.bodyMdMedium}>{t('amount')}</Label>
+        {/* The fiat value will always leave dust and is often inaccurate anyways */}
+        {!isFiatPrimary && <MaxClearButton asset={asset} />}
```

Comment:
Found a bug:
1. Select account that has no ETH balance
2. Click Max
3. Negative transfer amount is shown

I think we should disable Max if the user has no balance for the selected token

Reply:
Great catch!